R/size.R

```diff
@@ -48,14 +46,20 @@ vec_size <- function(x) {
 
 #' @export
 #' @rdname vec_size
-vec_size_common <- function(..., .size = NULL) {
+vec_size_common <- function(..., .size = NULL, .empty) {
```

Comment:
This either violates https://principles.tidyverse.org/args-required.html or https://principles.tidyverse.org/args-data-details.html — I think the easiest fix would be `.empty = stop_empty()` 

```R
stop_empty <- function() {
  abort("`...` is empty, and no `.empty` value was supplied.")
}
```

Reply:
This brings up another point. Should `stop_empty()` be exported? I've always felt that if a function is the default of an argument, a user should be able to call it. Which feels like part of https://principles.tidyverse.org/args-magical-defaults.html

Reply:
FWIW I don't think `.empty = NULL` is really a valid value here since it violates type stability.

I have a very rough start on exporting default helpers at https://principles.tidyverse.org/args-defaults-short.html#helper-function. I'll have to think about how to crosslink different problems that have the same solutions.

Reply:
Perhaps it doesn't violate type stability because the type can be inferred from the arguments statically. In statical typing terms, the type of the function is parameterised in its return value (the return value becomes an option type with the option depending on the declared type parameter).

So I think it is type-stable as long as we pass a constant.

Reply:
(or an expression whose type is constant, e.g. `new_foo()` or `dbl()`)

Reply:
For now I have:
- Set `.empty = NULL` as the default, which errors if no `...` are provided
- If provided, validate that `.empty` is a length 1 integer with `vec_assert()`

Reply:
I like that. Despite my notes above, there's no reason to allow the return type to be more complex here, since the caller can set a `na_int` sentinel if needed.

My only concern is the name, it seems `.empty` might suggest this is the size used for empty vectors. Should it be `.absent` instead?

Reply:
Updated to `.absent`